service/controller/resource/cloudconfigblob/create.go

```diff
@@ -0,0 +1,143 @@
+package cloudconfigblob
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-04-01/storage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	corev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/pingcap/errors"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-operator/v4/service/controller/blobclient"
+	"github.com/giantswarm/azure-operator/v4/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	azureMachinePool, err := key.ToAzureMachinePool(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	credentialSecret, err := r.getCredentialSecret(ctx, &azureMachinePool)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	blobName := key.BlobName(&azureMachinePool, key.PrefixWorker())
+
+	var payload string
+	{
+		payload, err = r.getCloudConfigFromBootstrapSecret(ctx, azureMachinePool)
+		if errors.IsNotFound(err) {
```

Comment:
If required objects are not created yet, or bootstrap CR is not ready, we cancel the current resource/handler, hoping another handler will fix that.

Reply:
Do you need to actually cancel the reconciliation of this resource or is returning just enough?

Reply:
We can:
- Return, which cancels this resource reconciliation, allowing the controller to continue to the next resource
- Set reconciliation canceled, which makes the controller to stop until next reconciliation loop, avoiding the controller to continue to the next resource

I believe in this case we want the first one. At least while we don't have any more resources in this controller.

Reply:
Nah, I think you are right. I'm going to cancel the whole reconciliation loop. We can't do anything if cloudconfig is not rendered.